Core/include/Acts/Surfaces/TrapezoidBounds.hpp

```diff
@@ -54,14 +55,29 @@ class TrapezoidBounds : public PlanarBounds {
     checkConsistency();
   }
 
+  /// Constructor for symmetric Trapezoid with rotation
+  ///
+  /// @param halfXnegY minimal half length X, definition at negative Y
+  /// @param halfXposY maximal half length X, definition at positive Y
+  /// @param halfY half length Y - defined at x=0
+  /// @param rotAngle: rotation angle of the bounds w.r.t coordinate axes
+  TrapezoidBounds(double halfXnegY, double halfXposY, double halfY,
+                  double rotAngle) noexcept(false)
+      : m_values({halfXnegY, halfXposY, halfY, rotAngle}),
+        m_boundingBox(std::max(halfXnegY, halfXposY), halfY),
+        m_rotMat{Eigen::Rotation2D<double>(rotAngle)} {
+    checkConsistency();
+  }
+
   /// Constructor for symmetric Trapezoid - from fixed size array
   ///
   /// @param values the values to be stream in
   TrapezoidBounds(const std::array<double, eSize>& values) noexcept(false)
       : m_values(values),
         m_boundingBox(
             std::max(values[eHalfLengthXnegY], values[eHalfLengthXposY]),
```

Comment:
I think the bounding box calculation needs to be updated. Right now it will still only consider the original unrotated trapezoid, right?

Reply:
Right but the bounding box is publicly accessible with a getter. I think the assumption is that the bounding box is in the native surface coordinate system, so I think it'd need to be rotated.

Reply:
can you implement this @junggjo9 ?